### PR TITLE
Setup server structure

### PR DESCRIPTION
**Pull request by JefZheng**
# Pull Request

## Description

- Added basic express server and setup server file structure. Also added some node packages needed to run express.
- Will now delete the bundled directly and rebuild every time a webpack script is called
- Added eslintrc and prettierrc json file
- Added babelrc file

## Type of change
Adding files, folders and packages
Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

1. Run `npm install`.
2. Run `npm run dev`
3. In a separate terminal run `npm run start`

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by davidzhangnyc — APPROVED** [review]
LGTM!

**Comment by dlande000** [comment]
@JefZheng your PR should tell me what to test and how. What commands should I run? Am I testing things in Chrome or the terminal or Electron? What exactly should I be looking for??

File: server/server.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+const app = express();
+const PORT = process.env.PORT || 3000;
+
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+
+app.use(express.static(path.resolve(__dirname + "../src")));
+app.use("/build", express.static(path.join(__dirname + "../build")));
+
+app.get("/", (req, res) => res.sendFile(path.join(__dirname, "../index.html")));
```
Comment (dlande000) [inline server/server.js]: Just me not knowing, but what's the distinction between `express.static` on line 11 and `res.sendFile` on 13? I understand the actual difference between those methods, but is there a reason to prefer the latter on 13 over the former?

File: server/server.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+const app = express();
+const PORT = process.env.PORT || 3000;
+
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+
+app.use(express.static(path.resolve(__dirname + "../src")));
+app.use("/build", express.static(path.join(__dirname + "../build")));
+
```
Comment (dlande000) [inline server/server.js]: Remove blank line.

File: server/server.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+app.get("/", (req, res) => res.sendFile(path.join(__dirname, "../index.html")));
+app.use((req, res) => res.status(404).send("Unable to find item"));
```
Comment (dlande000) [inline server/server.js]: Add blank line above.

File: server/server.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+app.use((err, req, res, next) => {
+  const defaultErr = {
+    log: "Express error handler caught unknown middleware error",
+    status: 500,
+    message: { err: "An error occurred" },
+  };
+  const errorObj = Object.assign({}, defaultErr, err);
```
Comment (dlande000) [inline server/server.js:21]: Add blank line after 21.

File: server/server.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  console.log(errorObj.log);
+  return res.status(errorObj.status).json(errorObj.message);
```
Comment (dlande000) [inline server/server.js:24]: add blank line after 23.

File: server/server.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+});
+
+app.listen(PORT, () => {
+  console.log(`Server running on ${PORT}...`);
```
Comment (dlande000) [inline server/server.js:28]: Add spaces: `${ PORT }`

**Review by dlande000 — CHANGES_REQUESTED** [review]
Code looks good—all requested changes are style-based. Please add instructions for how to test this PR and I will rereview.

File: server/server.js (hunk 1 above)
Comment (JefZheng) [reply server/server.js]: Now that I think about it, it is redundant to have two ways to serve up static files. Line 13 has been removed.

**Review by dlande000 — APPROVED** [review]

**Review by Hydroelectric29 — APPROVED** [review]

**Pull request merged**